### DISCUSSION TITLE
How to update a static page properly?

### DISCUSSION THREAD

**Discussion by kylemiller3** [vercel/next.js#20153] [Help]
I would love to keep statically rendered pages but I am having a problem updating the page on the client side. I have tried to force re render the page but it doesn't work using useEffect and setState. I noticed some components do refresh the DOM such as my Material UI drawer so I know it is possible at least. My use case is that I have a dynamic login indicator and that's the only thing that needs to be updated.

**Comment by jamesmosier** [comment]
Hi @kylemiller3. Do you have an example of what is not working? All Next.js pages are hydrated upon rendering in your browser at which time all React event handlers are attached.

**Reply by kylemiller3** [reply]
So the hydration process should have called and updated the client when finished? I worked around this by using a custom hook useUser that worked and refreshed the page when the user becomes available from auth. I'm not sure I can provide a concise example of what I'm talking about but I had used another custom useAuth in my app bar component and that had not worked to refresh the page.

I don't really understand how the hydration process works on the client side. Thanks for your help.

**Comment by kylemiller3** [comment]
I think my problem was a react-redux wrapper I was using
